src/components/DeleteModal.tsx

```diff
@@ -0,0 +1,44 @@
+import { Button } from 'antd';
+import classes from '../containers/wrapper/Wrapper.module.css';
+import { Contact } from '../interfaces/contacts';
+import { Item } from '../interfaces/tickets';
+
+export default function EditModal({
+  value,
+  setIsDeleteModalOpen,
+  handleDelete,
+  itemToDelete,
+  setItemToMakeAction,
+}: {
+  value: string;
+  setIsDeleteModalOpen: (val: boolean) => void;
+  handleDelete: (val: string) => void;
+  itemToDelete: Item | Contact | null;
+  setItemToMakeAction: (val: any) => void;
```

Comment:
```suggestion
  setItemToMakeAction: (val: Item | null) => void;
```
Or it's not working?

Reply:
Nope. DeleteModal is used in two components - Tickets and Contacts. Both of them path function setItemToMakeAction to DeleteModal as a prop, expecting the value to be (Item | null) for Tickets and (Contact | null) for Contact. So I thought setItemToMakeAction in DeleteModal could have type (Item | Contact | null), but that way both Tickets and Contacts fire an error, saying '
>Type 'Dispatch<SetStateAction<Contact | null>>' is not assignable to type '(val: Item | null) => void'' 

and vise-versa for Ticket component and Contact type. 

So as far as I understand, I can't use optional types that way, but I didn't figure out the solution.